managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -213,8 +213,11 @@ public void safeRun() {
             EntryImpl entry = EntryImpl.create(ledger.getId(), entryId, data);
             // EntryCache.insert: duplicates entry by allocating new entry and data. so, recycle entry after calling
             // insert
-            ml.entryCache.insert(entry);
-            entry.release();
+            // Entry cache doesn't copy the data if entry already exist into the cache.
+            // Backlog read tries to add entry into cache which can try to add duplicate entry into cache.
+            if (ml.entryCache.insert(entry)) {
+                entry.release();
+            }
```

Comment:
Here will be a potential memory leak? If the cache doesn't have enough space, will the entry not be released?

Reply:
I think @codelipenghui is right that this is a memory leak. I think that this part of the change should be reverted.

Reply:
I created #17007 to revert this.